File: benchmarks/simple-tests/ocamlcapi.ml
external test_no_args_alloc : unit -> int = "test_no_args_alloc"
external test_no_args_noalloc : unit -> int = "test_no_args_no_alloc" [@@noalloc]
external test_few_args_alloc : int -> int = "test_few_args_alloc"
external test_few_args_noalloc : int -> int = "test_few_args_no_alloc" [@@noalloc]
external test_many_args_alloc : int -> int -> int -> int -> int -> int -> int -> int = "test_many_args_noalloc_bc" "test_many_args_alloc_nc"
external test_many_args_noalloc : int -> int -> int -> int -> int -> int -> int -> int = "test_many_args_noalloc_bc" "test_many_args_noalloc_nc" [@@noalloc]

File: benchmarks/simple-tests/ocamlcapi.c
#include <stdio.h>
#include <caml/mlvalues.h>

CAMLprim value
test_no_args_alloc(value unit)
{
    return Int_val(42);
}

CAMLprim value test_no_args_no_alloc(value unit)
{
    return Int_val(42);
}

CAMLprim value test_few_args_alloc(value input) {
    return Int_val(42);
}

CAMLprim value test_few_args_no_alloc(value input) {
    return Int_val(42);
}

CAMLprim value test_many_args_alloc_nc(value one, value two, value three, value four, value five, value six, value seven) {
    return Int_val(42);
}

CAMLprim value test_many_args_noalloc_nc(value one, value two, value three, value four, value five, value six, value seven) {
    return Int_val(42);
}

